### ISSUE TITLE
[hashlink] threads are broken

### ISSUE THREAD

**Issue by RealyUniqueName** [HaxeFoundation/haxe#9875]
Looks like same thread handles are reused for multiple threads.
This issue blocks #9868 and #9111 for Hashlink.

**A sample to demonstrate messages being received in wrong threads:**
```haxe
import sys.thread.Thread;
import sys.thread.Mutex;

class Main {
	static function main() {
		var threads = [];
		var mutex = new Mutex();
		for(i in 0...10) {
			Thread.create(() -> {
				mutex.acquire();
				var thread = Thread.current();
				threads.push(thread);
				mutex.release();
				Sys.sleep(1);
				var msg = Thread.readMessage(false);
				if(msg != null) {
					trace('Received in $i-th thread');
				}
			});
		}
		Sys.sleep(0.5);
		// Sending messages to threads 7, 8 and 9
		threads[7].sendMessage('');
		threads[8].sendMessage('');
		threads[9].sendMessage('');
		Sys.sleep(1);
	}
}
```
Run it several times:
```
alex@alex-laptop:~/temp/test$ haxe --main Main --hl test.hl && hl test.hl
Main.hx:17: Received in 9-th thread
Main.hx:17: Received in 1-th thread
Main.hx:17: Received in 4-th thread
alex@alex-laptop:~/temp/test$ haxe --main Main --hl test.hl && hl test.hl
Main.hx:17: Received in 2-th thread
Main.hx:17: Received in 8-th thread
Main.hx:17: Received in 3-th thread
alex@alex-laptop:~/temp/test$ haxe --main Main --hl test.hl && hl test.hl
Main.hx:17: Received in 8-th thread
Main.hx:17: Received in 9-th thread
Main.hx:17: Received in 7-th thread
```

**A sample to demonstrate `Thread.current()` returns same value in different threads:**
```haxe
import sys.thread.Thread;
import sys.thread.Mutex;

class Main {
	static function main() {
		var threads = [];
		var mutex = new Mutex();
		for(i in 0...10) {
			Thread.create(() -> {
				mutex.acquire();
				var thread = Thread.current();
				for(t in threads) {
					if(t == thread) {
						trace('reused!');
						break;
					}
				}
				threads.push(thread);
				mutex.release();
			});
		}
	}
}
```
Run it multiple times:
```
alex@alex-laptop:~/temp/test$ haxe --main Main --hl test.hl && hl test.hl
Main.hx:14: reused!Main.hx:14: reused!
alex@alex-laptop:~/temp/test$ haxe --main Main --hl test.hl && hl test.hl
Main.hx:14: reused!
alex@alex-laptop:~/temp/test$ haxe --main Main --hl test.hl && hl test.hl
Main.hx:14: reused!
Main.hx:14: reused!Main.hx:14: reused!
alex@alex-laptop:~/temp/test$ haxe --main Main --hl test.hl && hl test.hl
Main.hx:14: reused!
```

**Comment by RealyUniqueName** [comment]
So, the issue is `pthread_create` reuses IDs for threads. And `sys.thread.Thread` implementation for HL does not drop references to finished threads.
I'll fix this in #9868

**Comment by RealyUniqueName** [comment]
My initial sample for messages was incorrect.
However it was possible to send a message to a thread, terminate that thread, and then start a new thread and receive that message in the new thread. This is fixed now.

**Comment by RealyUniqueName** [comment]
But `Thread.current()` still returns non-unique references, which will be fixed in #9868
